src/plugins/intel_cpu/src/emitters/plugin/x64/jit_bf16_emitters.hpp

```diff
@@ -33,56 +38,115 @@ class jit_uni_vcvtneps2bf16 : public jit_emitter {
+                Ymm out = Ymm(out_vec_idxs[0]);
+                saturate_input(truncated, in, "bf16_min", "bf16_max");
+                h->vcvtneps2bf16(out, truncated);
+            } else if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx512_core) {
+                saturate_input(truncated, in, "bf16_min", "bf16_max");
+
+                Zmm aux = Zmm(aux_vec_idxs[3]);
+                Zmm aux1 = Zmm(aux_vec_idxs[4]);
+                Ymm out = Ymm(out_vec_idxs[0]);
+
+                h->uni_vpsrld(aux, truncated, 16);
+                h->vpandd(aux, aux, table_val("one"));
+                h->uni_vmovups(aux1, table_val("even"));
+                h->uni_vpaddd(aux, aux1, aux);
+                h->uni_vpaddd(aux, truncated, aux);
+                h->vfixupimmps(aux, truncated, table_val("selector"), 0);
+                h->vpsrad(aux, aux, 16);
+                h->vpmovdw(out, aux);
+            } else if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::cpu_isa_t::avx2_vnni_2)) {
+                saturate_input(truncated, in, "bf16_min", "bf16_max");
+                Xmm out = Xmm(out_vec_idxs[0]);
+                h->vcvtneps2bf16(out, truncated, PreferredEncoding::VexEncoding);
             } else {
-                h->uni_vmovups(aux, in);
-                h->uni_vandps(aux, aux, table_val("rounding"));
+                Xmm out = Xmm(out_vec_idxs[0]);
+                Vmm aux = Vmm(aux_vec_idxs[3]);
+
+                saturate_input(truncated, in, "bf16_min", "bf16_max");
+
+                if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx2) {
+                    h->uni_vandps(aux, truncated, table_val("rounding"));
+                } else {
+                    h->uni_vmovups(aux, truncated);
+                    h->uni_vandps(aux, aux, table_val("rounding"));
+                }
+
+                h->uni_vpsrld(truncated, truncated, 16);
+                h->uni_vandps(truncated, truncated, table_val("mask_truncation_word"));
+                h->uni_vpackusdw(truncated, truncated, truncated);
+
+                if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx2) {
+                    h->vpermq(Ymm(truncated.getIdx()), Ymm(truncated.getIdx()), 0xD8);  // 11 01 10 00
+                    h->vextracti128(out, Ymm(truncated.getIdx()), 0);
+                } else {
+                    h->uni_vmovups(out, truncated);
+                }
             }
-
-            h->uni_vpsrld(aux, aux, 1);
-            h->uni_vpaddd(aux, aux, in);
-            h->uni_vpsrld(aux, aux, 16);
-
-            // dword to word using truncation
-            h->uni_vandps(aux, aux, table_val("mask_truncation_word"));
-            h->uni_vpackusdw(aux, aux, aux);
-
-            if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx2) {
-                h->vpermq(Ymm(aux.getIdx()), Ymm(aux.getIdx()), 0xD8); //11 01 10 00
-                h->vextracti128(out, Ymm(aux.getIdx()), 0);
+        } else {
```

Comment:
why `vcvtneps2bf16 `is not used while it's available, in truncation?

Reply:
discussed in the following comments.